lib/student.rb

```diff
@@ -0,0 +1,17 @@
+class Student
+    attr_reader :name, :age, :scores
+    def initialize(student_data)
+        @name = student_data[:name]
+        @age = student_data[:age]
+        @scores = student_data.fetch(:scores, [])
```

Comment:
We are doing a little bit too much here - looking closely at the spec, we never expect that scores will be passed to the student on creation, so there's no reason to have it grab the scores on initialization, we only add scores through the interface that we create. This is a great way to do the thing if we may or may not have scores upon creation but it falls under how we talked about earlier of lets not build something we dont need.